packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -237,26 +237,7 @@ export class OdspDocumentService implements IDocumentService {
                 ? Promise.resolve(null)
                 : this.getWebsocketToken!(options);
 
-            const joinSessionPromise = this.joinSession(requestWebsocketTokenFromJoinSession, options).catch((e) => {
-                const likelyFacetCodes = e as IFacetCodes;
-                if (Array.isArray(likelyFacetCodes.facetCodes)) {
-                    for (const code of likelyFacetCodes.facetCodes) {
-                        switch (code) {
-                            case "sessionForbiddenOnPreservedFiles":
-                            case "sessionForbiddenOnModerationEnabledLibrary":
-                            case "sessionForbiddenOnRequireCheckout":
-                                // This document can only be opened in storage-only mode.
-                                // DeltaManager will recognize this error
-                                // and load without a delta stream connection.
-                                this._policies = {...this._policies,storageOnly: true};
-                                throw new DeltaStreamConnectionForbiddenError(code, { driverVersion });
-                            default:
-                                continue;
-                        }
-                    }
-                }
-                throw e;
-            });
+            const joinSessionPromise = this.joinSession(requestWebsocketTokenFromJoinSession, options, false, true);
```

Comment:
maybe this is the right place to use config for supportJoinSessionRefresh

Reply:
Don't forget to pass "FluidRemoveCheckAccess" through the prefer header.
Pr reference: https://onedrive.visualstudio.com/SharePoint%20Online/_git/SPO/pullrequest/890197?_a=files&path=/sts/Vroom/src/Providers/OpStreamProvider.cs

Reply:
Passing the header.